packages/template-retail-react-app/jest-setup.js

```diff
@@ -10,6 +10,68 @@ require('raf/polyfill') // fix requestAnimationFrame issue with polyfill
 require('@testing-library/jest-dom/extend-expect')
 const {Crypto} = require('@peculiar/webcrypto')
 
+const {setupServer} = require('msw/node')
+const {rest} = require('msw')
+const {
+    mockCategory,
+    mockedRegisteredCustomer,
+    exampleTokenReponse
+} = require('./app/commerce-api/mock-data')
+
+const AJwtThatNeverExpires =
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiZXhwIjoyNjczOTExMjYxLCJpYXQiOjI2NzM5MDk0NjF9.BDAp9G8nmArdBqAbsE5GUWZ3fiv2LwQKClEFDCGIyy8'
+
+/**
+ * Set up an API mocking server for testing purposes.
+ * This mock server includes the basic oauth flow endpoints.
+ */
+export const setupMockServer = () => {
+    return setupServer(
+        rest.post('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.get('*/oauth2/authorize', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/login', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.get('*/oauth2/logout', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(exampleTokenReponse))
+        ),
+        rest.get('*/customers/:customerId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockedRegisteredCustomer))
+        ),
+        rest.post('*/sessions', (req, res, ctx) => res(ctx.delay(0), ctx.status(200))),
+        rest.post('*/oauth2/token', (req, res, ctx) =>
+            res(
+                ctx.delay(0),
+                ctx.json({
+                    customer_id: 'test',
+                    access_token: AJwtThatNeverExpires,
+                    refresh_token: 'testrefeshtoken',
+                    usid: 'testusid',
+                    enc_user_id: 'testEncUserId',
+                    id_token: 'testIdToken'
+                })
+            )
+        ),
+        rest.get('*/categories/:categoryId', (req, res, ctx) =>
+            res(ctx.delay(0), ctx.status(200), ctx.json(mockCategory))
+        ),
+        rest.post('*/baskets/actions/merge', (req, res, ctx) => res(ctx.delay(0), ctx.status(200)))
+    )
+}
+
+global.server = setupMockServer()
```

Comment:
Using `global` is a bad practice. We should always use `import`/`export` unless there's a strong reason not to (e.g. it completely breaks jest or something).

Reply:
I tried to use `import`/`export` by creating a global `mockServer` instance from the test-utils.js file and then control the lifecycle of the server from jest-setup.js, however, for some reason, this cause the mock overriding in individual tests to fail.